src/lib/player.rs

```diff
@@ -35,71 +39,72 @@ pub struct SpotifyPlayer {
 }
 
 pub struct EmittedSink {
-    sender: Arc<SyncSender<u8>>,
-    pub receiver: Arc<Mutex<Receiver<u8>>>,
+    sender: Arc<SyncSender<[f32; 2]>>,
+    pub receiver: Arc<Mutex<Receiver<[f32; 2]>>>,
+    input_buffer: Arc<Mutex<(Vec<f32>, Vec<f32>)>>,
+    resampler: Arc<Mutex<FftFixedInOut<f32>>>,
+    resampler_input_frames_needed: usize,
 }
 
 impl EmittedSink {
     fn new() -> EmittedSink {
-        let (sender, receiver) = sync_channel::<u8>(64);
+        let (sender, receiver) = sync_channel::<[f32; 2]>((1024f32 * 1.2).ceil() as usize);
```

Comment:
Is this saying that the upper bound is 1.2 * the amount of data in a single frame? If so, could you just add a comment explaining it?

Reply:
Yes. I added a comment explaining the meaning and replaced the vague calculation with the hardcoded value for our used chunk size and frequencies.